### DISCUSSION TITLE
Is it possible to embed code snippets?

### DISCUSSION THREAD

**Discussion by noamsiegel** [requarks/wiki#5581] [Help / Questions] [answered]
Hello all,

Before jumping into Wiki.js, I want to make sure that we can embed/reference real code, in real-time. For example, most people are comfy documenting like so:

  ```csharp
  public static void Log(string message)
  {
      _logger.LogInformation(message);
  }
  ```

But, we would like to reference a code block within a local/external repo:

`:::code language="csharp" source="intro/samples/cu/Controllers/StudentsController.cs" range="2-24,26":::`

Is this possible? Or does one have to jump back and forth between classes/functions to document them?

Thanks in advance, and for more info, [click here](https://docs.microsoft.com/en-us/contribute/code-in-docs).

**Comment by NGPixel** [comment, marked as answer]
Using the markdown editor, you can use code blocks, e.g.:

````````
```csharp
public static void Log(string message)
{
    _logger.LogInformation(message);
}
```
````````

**Reply by noamsiegel** [reply]
This is great, but I'd like to call specific code lines/functions/etc from within the actual src files. For example, I'd like to reference the function "example" from "file.py" and have it literally included inside the documentation. 

If there are changes to "example", then the referenced documentation inline code should change as well. 

Just wondering if this is possible, sorry for any confusion!

**Reply by NGPixel** [reply]
Highlighting specific lines is not possible at the moment, but is planned for a future release.
